packages/flutter_tools/lib/src/base/io.dart

```diff
@@ -209,15 +210,54 @@ class _PosixProcessSignal extends ProcessSignal {
 /// A class that wraps stdout, stderr, and stdin, and exposes the allowed
 /// operations.
 class Stdio {
-  const Stdio();
+  Stdio();
+
+  @visibleForTesting
+  Stdio.test({
+    @required io.Stdout stdout,
+    @required io.IOSink stderr,
+  }) : _stdoutOverride = stdout, _stderrOverride = stderr;
+
+  io.Stdout _stdout;
+  io.IOSink _stderr;
+
+  bool _stdoutDone = false;
+  bool _stderrDone = false;
+
+  io.Stdout _stdoutOverride;
+  io.IOSink _stderrOverride;
 
   Stream<List<int>> get stdin => io.stdin;
 
   @visibleForTesting
-  io.Stdout get stdout => io.stdout;
+  io.Stdout get stdout {
+    if (_stdout == null) {
+      _stdout = _stdoutOverride ?? io.stdout;
+      _stdout.done.then(
```

Comment:
nit: maybe use `void _` instead of `v` if we're not going to use it

Reply:
Done.